File: src/main/java/com/jss/osiris/modules/myjss/wordpress/service/JssCategoryService.java
package com.jss.osiris.modules.myjss.wordpress.service;

import java.util.List;

import com.jss.osiris.modules.myjss.wordpress.model.JssCategory;

public interface JssCategoryService {
        public JssCategory getJssCategory(Integer id);

        public List<JssCategory> getAvailableJssCategories();

        public JssCategory addOrUpdateJssCategory(JssCategory jssCategory);
}

File: client/src/app/modules/miscellaneous/services/jss.category.service.ts
import { HttpClient, HttpParams } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { AppRestService } from "src/app/services/appRest.service";
import { JssCategory } from "../model/JssCategory";

@Injectable({
  providedIn: 'root'
})
export class JssCategoryService extends AppRestService<JssCategory> {

  constructor(http: HttpClient) {
    super(http, "myjss/wordpress");
  }

  getJssCategories() {
    return this.getList(new HttpParams(), "jss-categories");
  }
}
